libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/ConfigurationMemory.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using AdaptiveExpressions.Memory;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Bot.Builder.Integration.Runtime
+{
+    /// <summary>
+    /// Provides a wrapper around <see cref="IConfiguration"/> that adheres to the <see cref="IMemory"/> interface,
+    /// enabling evaluation of adaptive expressions defined within the runtime settings against the constructed
+    /// application configuration object.
+    /// </summary>
+    internal class ConfigurationMemory : IMemory
+    {
+        private const string SupportedSeparator = ":";
+
+        private static readonly Regex UnsupportedSeparators = new Regex("(\\.|_)+");
+
+        private readonly IConfiguration _configuration;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConfigurationMemory"/> class.
+        /// </summary>
+        /// <param name="configuration">The <see cref="IConfiguration"/> instance to be wrapped.</param>
+        public ConfigurationMemory(IConfiguration configuration)
+        {
+            this._configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
```

Comment:
>this. [](start = 12, length = 5)

The dotnet guidelines adopted favor not using "this" unless necessary. Please remove all instances.